PlagiarismSubmissionSubmitListener.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace APP\plugins\generic\plagiarism;
+
+use PKP\observers\events\SubmissionSubmitted;
+use Illuminate\Events\Dispatcher;
+
+class PlagiarismSubmissionSubmitListener
+{
+
+    /** @var PlagiarismPlugin $plugin */
+    private $plugin;

```

Comment:
perhaps better to use exact property type hint like `private PlagiarismPlugin $plugin` ?

Reply:
Yes, throughout.